src/Func.h

```diff
@@ -2281,6 +2281,12 @@ class Func {
      */
     Func &async();
 
+    /** Adds an extra dimension to the function storage to enable
+     * double buffering. This is especially useful in combination
+     * with the \ref Func::async.
+     */
+    Func &ring_buffer(Expr extent);
+
```

Comment:
Is it legal to specify extent=0 or extent=1?

Reply:
Extent=1 should be fine, but useless, I've added a comment about expected extent value.